### PR TITLE
 [website] Render Pulsar REST API in ReDoc and Fix endpoint template

### PR DESCRIPTION
**Pull request by sijie**
### Motivation

[ReDoc](https://github.com/Rebilly/ReDoc) provides a good mechanism to load OpenAPI/Swagger-generated API reference documentation.

 ### Changes

- Bump the swagger dependencies
- Exclude v1 APIs
- Exclude functions endpoints (it is using a lot of memories to load function endpoints)
- Use ReDoc to load the pulsar swagger definition
- Fix the rest admin api references and `endpoint` macros

### CODE REVIEW

**Comment by sijie** [comment]
@cckellogg ^^

File: src/gen-swagger.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+PULSAR_PATH=$(git rev-parse --show-toplevel)
+
+cd $PULSAR_PATH
+
+echo "Generating swagger json file ..."
+mvn -am -pl pulsar-broker install -DskipTests -Pswagger
+echo "Swagger json file is generated."
+
+cp pulsar-broker/target/docs/swagger.json site2/website/static/swagger/
```
Comment (cckellogg) [inline src/gen-swagger.sh]: will this file be auto generated when we build the site or will we build it and check it in?

File: site2/website/package.json [hunk 2]
```diff
@@ -20,5 +20,8 @@
   },
   "jest": {
     "preset": "jest-puppeteer"
+  },
+  "dependencies": {
+    "redoc": "^2.0.0-alpha.30"
```
Comment (cckellogg) [inline site2/website/package.json]: Can this be moved to devDependencies since the site is statically generated.

Comment (sijie) [reply site2/website/package.json]: yeah moved

File: src/gen-swagger.sh (hunk 1 above)
Comment (sijie) [reply src/gen-swagger.sh]: yes. it will be auto generated when we build the site. I am adding this script here but it will be called in the CI build.

**Comment by sijie** [comment]
rerun cpp tests

**Comment by sijie** [comment]
rerun cpp tests

**Pull request merged**